---
title: "YARN_debug"
author: "Ursula Widocki"
date: "5/13/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(tidyverse)
library(data.table)
library(yarn)
library(downloader)
library(biomaRt)
```


# trying this the YARN way
```{r}
# function to download the data directly from GTEx
#youLoadGTEx <- function(){
phenoFile <- "https://storage.googleapis.com/gtex_analysis_v8/annotations/GTEx_Analysis_v8_Annotations_SampleAttributesDS.txt"
pheno2File <- "https://storage.googleapis.com/gtex_analysis_v8/annotations/GTEx_Analysis_v8_Annotations_SubjectPhenotypesDS.txt"
#geneFile <- "https://storage.googleapis.com/gtex_analysis_v8/rna_seq_data/GTEx_Analysis_2017-06-05_v8_RNASeQCv1.1.9_gene_reads.gct.gz"

message("Downloading and reading files")
pdFile <- tempfile("phenodat", fileext = ".txt")
download(phenoFile, destfile = pdFile)
pd <- read_tsv(pdFile)
pd <- as.matrix(pd)
rownames(pd) <- pd[, "SAMPID"]
ids <- sapply(strsplit(pd[, "SAMPID"], "-"), function(i) paste(i[1:2], collapse = "-"))

pd2File <- tempfile("phenodat2", fileext = ".txt")
download(pheno2File, destfile = pd2File)
pd2 <- read_tsv(pd2File)
pd2 <- as.matrix(pd2)
rownames(pd2) <- pd2[, "SUBJID"]
pd2 <- pd2[which(rownames(pd2) %in% unique(ids)), ]
pd2 <- pd2[match(ids, rownames(pd2)), ]
rownames(pd2) <- colnames(counts)

pdfinal <- AnnotatedDataFrame(data.frame(cbind(pd, pd2)))

cnts <- fread("/Users/ursulawidocki/Desktop/BarabasiLab/Data/GTEx_Analysis_2017-06-05_v8_RNASeQCv1.1.9_gene_reads.gct")

# get sum expression of repeated genes
repeat_genes <- unique(cnts$Description[duplicated(cnts$Description)])
repeat_genes_agg = aggregate(.~Description, cnts[cnts$Description %in% repeat_genes,], sum)
cnts = rbind(data_df[!(cnts$Description %in% repeated_genes),], repeat_genes_agg)

genes <- unlist(cnts[, 1])
geneNames <- unlist(cnts[, 2])
# <---- insert get sum values of repeated genes
counts <- cnts[, -c(1:2)]
counts <- as.matrix(counts)
rownames(counts) <- genesNames
for (i in 1:nrow(problems(cnts))) {
  counts[problems(cnts)$row[i], problems(cnts)$col[i]] <- 1e+05
}
# remove genes with 0 count
throwAway <- which(rowSums(counts) == 0) 
counts <- counts[-throwAway, ]
genes <- sub("\\..*", "", rownames(counts))

#host <- "dec2013.archive.ensembl.org" # this needs to be updated
host <- "http://dec2021.archive.ensembl.org"
biomart <- "ENSEMBL_MART_ENSEMBL"
dataset <- "hsapiens_gene_ensembl"
attributes <- c("ensembl_gene_id", "hgnc_symbol", "chromosome_name",
                "start_position", "end_position", "gene_biotype")

message("Creating ExpressionSet")
pdfinal <- pdfinal[match(colnames(counts), rownames(pdfinal)),
]
es <- ExpressionSet(as.matrix(counts))
phenoData(es) <- pdfinal
#pData(es)["GTEX-YF7O-2326-101833-SM-5CVN9", "SMTS"] <- "Skin"
#pData(es)["GTEX-YEC3-1426-101806-SM-5PNXX", "SMTS"] <- "Stomach"

message("Annotating from biomaRt")
es <- annotateFromBiomart(obj = es, genes = genesNames, host = host,
                          biomart = biomart, dataset = dataset, attributes = attributes)

message("Cleaning up files")
unlink(pdFile)
unlink(pd2File)
unlink(countsFile)

#if (!is.null(file))
#saveRDS(es, file = file)
#return(es)
#}

```

```{r}
test<- youLoadGTEx()
```

Normalize gene counts
```{r}
tiss_norm <- normalizeTissueAware(es, groups = "SMTSD", normalizationMethod = "quantile")
dim(es)
dim(tiss_norm)
```

Filter lowly expressed genes
```{r}
es_filtered = yarn::filterLowGenes(tiss_norm,"SMTSD")
#dim(es)
dim(es_filtered)
```

Differential Expression
```{r}

```










